tests/test_views.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_raise_404_for_anonymous_user(self):
+        """
+        Test the validation for anonymous users.
+        """
+        user = AnonymousUser()
+        self.request.user = user
+        view = RefreshCourseSkills()
+        view.setup(self.request)
+        with self.assertRaises(Http404):
+            view.get_context_data()
+
+    @mock.patch('taxonomy.views.RefreshCourseSkills._update_course_skills')
+    @mock.patch('taxonomy.views.render')
+    def test_form_post(self, get_course_skills_mock, mock_render):
+        """
+        Test the data in form on post request.
+        """
+        request = self.request
+        course_uuid = 'uuid-1234'
+        request.user = self.user
+        get_course_skills_mock.return_value = {}
+        request.POST = {'course_uuid': course_uuid}
+        view = RefreshCourseSkills()
+        view.setup(request)
+        view.post(request)
+        self.assertEqual(mock_render.call_args[0][0], course_uuid)
+
+    @mock.patch('taxonomy.views.RefreshCourseSkills._update_course_skills')
+    @mock.patch('taxonomy.views.render')
+    def test_form_post_empty_data(self, get_course_skills_mock, mock_render):
+        """
+        Test the empty data in form on post request.
+        """
+        request = self.request
+        course_uuid = ''
+        request.user = self.user
+        get_course_skills_mock.return_value = {}
+        request.POST = {'course_uuid': course_uuid}
+        view = RefreshCourseSkills()
+        view.setup(request)
+        view.post(request)
+        self.assertEqual(mock_render.call_args, None)
+
+    @mock.patch('taxonomy.management.commands.refresh_course_skills.get_course_metadata_provider')
+    @mock.patch('taxonomy.management.commands.refresh_course_skills.EMSISkillsApiClient.get_course_skills')
+    def test_update_course_skill(self, get_course_skills_mock, get_course_provider_mock):
+        """
+        Test that the course skills are updated on passing valid course uuid.
+        """
+        request = self.request
+        # Annotate a request object with a session
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
+
+        # Annotate a request object with a messages
+        middleware = MessageMiddleware()
+        middleware.process_request(request)
+        request.session.save()
+
+        get_course_skills_mock.return_value = SKILLS
+        course = MockCourse()
+        get_course_provider_mock.return_value = DiscoveryCourseMetadataProvider([course])
+        request.user = self.user
+        view = RefreshCourseSkills()
+        view.setup(request)
+        view._update_course_skills(course.uuid)  # pylint: disable=protected-access
+
+        msg = ('Skills data for course_key: {} is updated.'.format(course.uuid))
+        msg_tag = 'success'
+        storage = messages.get_messages(request)
+        for message in storage:
+            print(message.message)
```

Comment:
Can we remove this statement? I'm assuming that you used it for testing purposes maybe

Reply:
Good catch 💯